### PR TITLE
add cosine distance

### PR DESCRIPTION
**Pull request by kmrmt**
Cosine distance is enabled to use from NGT C API.
I enabled cosine distance in gongt.

please review

### CODE REVIEW

File: gongt.go [hunk 1]
```diff
@@ -377,7 +381,15 @@ func (n *NGT) Open() *NGT {
...
 		}
 	case Cosine:
-		// TODO: not exists C API
+		if C.ngt_set_property_distance_type_cosine(prop, ebuf) == ErrorCode {
+			n.errs = append(n.errs, newGoError(ebuf))
+			return n
+		}
+	case NormalizedAngle:
+		// TODO: not exists in C API
+		return n
```
Comment (kpango) [inline gongt.go:394]: I think append `not implement error` is better

Comment (kmrmt) [reply gongt.go:394]: Thank you!
I implemented returning error.

File: gongt.go [hunk 2]
```diff
@@ -386,10 +386,12 @@ func (n *NGT) Open() *NGT {
 			return n
 		}
 	case NormalizedAngle:
-		// TODO: not exists in C API
+		// TODO: not implemented in C API
+		n.errs = append(n.errs, errors.New("Not implemented in C API"))
```
Comment (kpango) [inline gongt.go]: Can you define Error Variable?
`ErrCAPINotImplemented := errors.New("Not implemented in C API")`

**Review by kpango — APPROVED** [review]
LGTM

**Pull request merged**